Xamarin.Forms.Core/Visuals/VisualTypeConverter.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Xamarin.Forms
+{
+	[Xaml.TypeConversion(typeof(IVisual))]
+	public class VisualTypeConverter : TypeConverter
+	{
+		static Dictionary<string, IVisual> _visualTypeMappings;
+		static Dictionary<string, IVisual> InitMappings()
+		{
+			var mappings = new Dictionary<string, IVisual>(StringComparer.OrdinalIgnoreCase);
+			_visualTypeMappings = mappings;
+			return mappings;
+		}
+
+		public static void Register(string visualKey, IVisual visual)
+		{
+			if (_visualTypeMappings == null)
+				InitMappings();
+
+			_visualTypeMappings[visualKey] = visual;
+		}
+
+		internal static void Register(Type visual)
+		{
+			if (_visualTypeMappings == null)
+				InitMappings();
+
+			if(!_visualTypeMappings.ContainsKey(visual.FullName))
+			{
+				try
+				{
+					IVisual registeredVisual = (IVisual)Activator.CreateInstance(visual);
+					string name = visual.Name;
+					string fullName = visual.FullName;
+
+					if (name.EndsWith("Visual", StringComparison.OrdinalIgnoreCase))
+					{
+						name = name.Substring(0, name.Length - 6);
+						fullName = fullName.Substring(0, fullName.Length - 6);
+					}
+
+					_visualTypeMappings[name] = registeredVisual;
+					_visualTypeMappings[fullName] = registeredVisual;
+					_visualTypeMappings[$"{name}Visual"] = registeredVisual;
+					_visualTypeMappings[$"{fullName}Visual"] = registeredVisual;
```

Comment:
With the rename of the visual to XxxVisual, we probably have to check the reverse: 
if endsWith("Visual"), add (name-"Visual)